gabby/iready/view/tests_by_round.sql

```diff
@@ -0,0 +1,155 @@
+  iready.tests_by_round AS
+WITH
+  iready AS (
+    SELECT
+      LEFT(dr.academic_year, 4) AS academic_year,
+      dr.student_id,
+      co.region,
+      co.school_abbreviation,
+      co.grade_level,
+      CASE
+        WHEN dr._file LIKE '%ela%' THEN 'Reading'
+        WHEN dr._file LIKE '%math%' THEN 'Math'
+      END AS [subject],
+      dr.start_date,
+      dr.completion_date,
+      RIGHT(rt.time_per_name, 1) AS round_number,
+      COALESCE(rt.alt_name, 'Outside Round') AS test_round,
+      CASE
+        WHEN rt.alt_name = 'BOY' THEN 'Fall ' + LEFT(dr.academic_year, 4)
+        WHEN rt.alt_name = 'MOY' THEN 'Winter ' + RIGHT(dr.academic_year, 4)
+        WHEN rt.alt_name = 'EOY' THEN 'Spring ' + RIGHT(dr.academic_year, 4)
+      END AS test_round_date,
+      dr.baseline_diagnostic_y_n_,
+      dr.most_recent_diagnostic_y_n_,
+      dr.overall_scale_score,
+      dr.[percentile],
+      dr.overall_relative_placement,
+      CASE
+        WHEN dr.overall_relative_placement = '3 or More Grade Levels Below' THEN 1
+        WHEN dr.overall_relative_placement = '2 Grade Levels Below' THEN 2
+        WHEN dr.overall_relative_placement = '1 Grade Level Below' THEN 3
+        WHEN dr.overall_relative_placement = 'Early On Grade Level' THEN 4
+        WHEN dr.overall_relative_placement = 'Mid or Above Grade Level' THEN 5
+      END AS orp_numerical,
+      CASE
+        WHEN dr.overall_relative_placement IN (
+          'Early On Grade Level',
+          'Mid or Above Grade Level'
+        ) THEN 'On or Above Grade Level' -- trunk-ignore(sqlfluff/LT05)
+        -- trunk-ignore(sqlfluff/LT05)
+        WHEN dr.overall_relative_placement = '1 Grade Level Below' THEN dr.overall_relative_placement
+        WHEN dr.overall_relative_placement IN (
+          '2 Grade Levels Below',
+          '3 or More Grade Levels Below'
+        ) THEN 'Two or More Grade Levels Below'
+      END AS placement_3_level,
+      dr.rush_flag,
+      dr.mid_on_grade_level_scale_score,
+      dr.percent_progress_to_annual_typical_growth_,
+      dr.percent_progress_to_annual_stretch_growth_,
+      dr.diagnostic_gain,
+      dr.annual_typical_growth_measure,
+      dr.annual_stretch_growth_measure,
+      COUNT(*) OVER (
+        PARTITION BY
+          rt.alt_name,
+          dr.student_id,
+          dr.academic_year,
+          dr._file
+        ORDER BY
+          dr.completion_date DESC
+      ) AS rn_subj_round,
+      COUNT(*) OVER (
+        PARTITION BY
+          dr.student_id,
+          dr.academic_year,
+          dr._file
+        ORDER BY
+          dr.completion_date DESC
+      ) AS rn_subj_year
+    FROM
+      gabby.iready.diagnostic_results AS dr
+      LEFT JOIN gabby.people.school_crosswalk AS sc ON (dr.school = sc.site_name)
+      LEFT JOIN gabby.reporting.reporting_terms AS rt ON (
+        (
+          dr.completion_date BETWEEN rt.start_date AND rt.end_date
+        )
+        AND rt.identifier = 'IR'
+        AND sc.region = rt.region
```

Comment:
is this 1 RT record per region?

Reply:
3 rounds, one record per region, per year